src/main/java/reposense/model/FileType.java

```diff
@@ -0,0 +1,75 @@
+package reposense.model;
+
+import java.nio.file.FileSystems;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a file type for use in {@link FileTypeManager}.
+ */
+public class FileType {
+    private static final String FILE_TYPE_VALIDATION_REGEX = "[A-Za-z0-9]+";
+    private static final String MESSAGE_ILLEGAL_FILE_TYPE = "The provided file type, %s, contains illegal characters.";
+
+    private String label;
+    private PathMatcher pathsGlob;
+
+    public FileType(String label, List<String> paths) {
+        validateFileType(label);
+        this.label = label;
```

Comment:
The name of `label` can be decided, maybe `name` or `type`? Can ask seniors' opinion.

Reply:
label itself or name are fine with me, can consult others' opinion too.